### PR TITLE
Add support for anchor redirects with client-side js

### PR DESCRIPTION
**Pull request by zanieb**
The redirect plugin doesn't support this, and it's not feasible to do server-side so we need to do the redirect client-side with some javascript.

### CODE REVIEW

File: docs/js/extra.js [hunk 1]
```diff
@@ -8,8 +8,8 @@ function cleanupClipboardText(targetSelector) {
     .filter(
       (node) =>
         !excludedClasses.some((className) =>
-          node?.classList?.contains(className),
-        ),
+          node?.classList?.contains(className)
+        )
```
Comment (zanieb) [inline docs/js/extra.js:12]: Prettier strikes!

File: docs/js/extra.js [hunk 2]
```diff
@@ -51,3 +51,42 @@ function setCopyText() {
...
+(function () {
+  let redirect_maps = {
+    "concepts/projects/#managing-dependencies":
+      "concepts/projects/dependencies/",
+  };
+
+  // The prefix for the site, see `site_dir` in `mkdocs.yml`
+  let site_dir = "uv";
+
+  function get_path() {
```
Comment (dhruvmanila) [inline docs/js/extra.js:66]: Is this required? Could we just do `window.location.pathname + window.location.hash` ?

File: docs/js/extra.js [hunk 3]
```diff
@@ -51,3 +51,42 @@ function setCopyText() {
 document$.subscribe(function () {
   setCopyText();
 });
+
+// Use client-side redirects for anchors that have moved.
+// Other redirects should use `redirect_maps` in the `mkdocs.yml` file instead.
+(function () {
```
Comment (dhruvmanila) [inline docs/js/extra.js:57]: We'll probably need to add an event listener for this, mostly on [`DOMContentLoaded`](https://developer.mozilla.org/en-US/docs/Web/API/Document/DOMContentLoaded_event).

**Review by dhruvmanila — APPROVED** [review]

File: docs/js/extra.js (hunk 2 above)
Comment (zanieb) [reply docs/js/extra.js:66]: Is the pathname guaranteed to end in a `/`?

File: docs/js/extra.js (hunk 3 above)
Comment (zanieb) [reply docs/js/extra.js:57]: Why? I don't think you can get to a page that doesn't exist except on initial load — i.e., like only an external link should reference it.

File: docs/js/extra.js (hunk 2 above)
Comment (dhruvmanila) [reply docs/js/extra.js:86]: I'm not sure, I can't find any documentation about whether this is guaranteed or not but from my experience modern browsers do tend to maintain this consistency even if you might not include a trailing `/` the browser will include it. I think it's fine to keep this in just to be on the safer side.

File: docs/js/extra.js (hunk 3 above)
Comment (dhruvmanila) [reply docs/js/extra.js:57]: Oh right, because all the internal references have been updated.

**Pull request merged**